INSTALL.md

```diff
@@ -0,0 +1,130 @@
+```sh
+$ go get github.com/golang/dep
+$ go install github.com/golang/dep/cmd
+
+# Make sure you have a go file in your directory which imports k8s.io/client-go
+# first--I suggest copying one of the examples.
+$ dep ensure k8s.io/client-go@^1.5.1
+```
+
+This will set up a /vendor directory in your current directory, add `k8s.io/client-go`
+to it, and flatten all of `k8s.io/client-go`'s dependencies into that vendor directory,
+so that your code and `client-go` will both get the same copy of each
+dependency.
+
+After installing like this, you could either use dep for your other
+dependencies, or copy everything in the `vendor` directory into your
+`$GOPATH/src` directory and proceed as if you had done a fancy `go get` that
+flattened dependencies sanely.
+
+One thing to note about dep is that it will omit dependencies that aren't
+actually used, and some dependencies of `client-go` are used only if you import
+one of the plugins (for example, the auth plugins). So you may need to run `dep
+ensure` again if you start importing a plugin that you weren't using before.
+
+### Godep
+
+[godep](github.com/tools/godep) is an older dependency management tool, which is
+used by the main Kubernetes repo and `client-go` to manage dependencies.
+
+Before proceeding with the below instructions, you should ensure that your
+$GOPATH is empty except for containing your own package and its dependencies,
+and you have a copy of godep somewhere in your $PATH.
+
+To install `client-go` and place its dependencies in your `$GOPATH`:
+
+```sh
+go get k8s.io/client-go
+cd $GOPATH/src/k8s.io/client-go
+git checkout v1.5.1
+cd 1.5 # only necessary with 1.5 and 1.4 clients.
+godep restore ./...
+```
+
+At this point, `client-go`'s dependencies have been placed in your $GOPATH, but
+if you were to build, `client-go` would still see its own copy of its
+dependencies in its `vendor` directory. You have two options at this point.
+
+If you would like to keep dependencies in your own project's vendor directory,
+then you can continue like this:
+
+```sh
+cd $GOPATH/src/<my-pkg>
+godep save ./...
+```
+
+Alternatively, if you want to build using the dependencsie in your `$GOPATH`,
+then `rm -rf vendor/` to remove `client-go`'s copy of its dependencies.
+
+### Glide
+
+[glide](https://github.com/Masterminds/glide) is another popular dependency
+management tool for go. Glide will manage your /vendor directory, but unlike
+godep, will not modify your $GOPATH (there's no equivalent of `godep restore` or
+`godep save`). To get the `client-go` dependency list into glide's format, you
+can run this series of commands:
+
+```sh
+go get k8s.io/client-go
+cd $GOPATH/src/k8s.io/client-go
+git checkout v1.5.1
+cd 1.5 # only necessary with 1.5 and 1.4 clients.
+glide init # Will import from godep format and produce glide.yaml
+```
+
+At this point, you could copy the `glide.yaml` file into your own project and
+run `glide up`. Alternatively, if you already have a `glide.yaml` file, you'll
```

Comment:
This doesn't look right. The `glide.yaml` won't contain any entry for client-go, but "my own project" depends on client-go. Could you double check?

Sorry, I can't verify if this works. The redirection of gopkg.in doesn't work for me, maybe it's because on mac, or because i'm in China..